src/events/processCollectedSignatures/estimateGas.js

```diff
@@ -0,0 +1,51 @@
+const { HttpListProviderError } = require('http-list-provider')
+const {
+  AlreadyProcessedError,
+  IncompatibleContractError,
+  InvalidValidatorError
+} = require('../../utils/errors')
+const { parseMessage } = require('../../utils/message')
+
+async function estimateGas({
+  foreignBridge,
+  validatorContract,
+  message,
+  numberOfCollectedSignatures,
+  v,
+  r,
+  s,
+  address
+}) {
+  try {
+    const gasEstimate = await foreignBridge.methods
+      .executeSignatures(v, r, s, message)
+      .estimateGas()
+    return gasEstimate
+  } catch (e) {
+    if (e instanceof HttpListProviderError) {
+      throw e
+    }
+
+    const requiredSignatures = await foreignBridge.methods.requiredSignatures().call()
```

Comment:
I don't really understand how it works. `requiredSignatures` is a method from Validators contract whereas `foreignBridge` contract is used here.

Reply:
This was a mistake. I must've missed this when I tested this manually. Thanks, fixed.